src/qtiCommonRenderer/renderers/interactions/GraphicGapMatchInteraction.js

```diff
@@ -126,62 +132,103 @@ const _shapesUnSelectable = function _shapesUnSelectable(interaction) {
-        matchMax = +choice.attr('matchMax');
-        usages = +_choiceUsages[choiceSerial];
+    // disable gapfiller if maxium usage reached
+    if (!interaction.responseMappingMode && gapFiller.attr('matchMax')) {
+        matchMax = +gapFiller.attr('matchMax');
+        usages = +_gapFillerUsages[gapFillerSerial];
 
-        // note: if matchMax is 0, then test taker is allowed unlimited usage of that choice
+        // note: if matchMax is 0, then test taker is allowed unlimited usage of that gapfiller
         if (matchMax !== 0 && matchMax <= usages) {
-            $choice.addClass('disabled');
-            $choice.removeClass('selectable');
-            $choice.find('img').attr('draggable', 'false'); //prevent native drag of image (Chrome, mouse)
+            $gapFiller.addClass('disabled');
+            $gapFiller.removeClass('selectable');
+            $gapFiller.find('img').attr('draggable', 'false'); //prevent native drag of image (Chrome, mouse)
         }
     }
 };
 
 /**
- * Unset a choice and unmark as disabled
+ * Unset a gapfiller (= image) and unmark as disabled
  * @private
  * @param {Object} interaction
- * @param {JQuery Element} $choice
+ * @param {JQuery Element} $gapFiller
  */
-const _unsetChoice = function _unsetChoice(interaction, $choice) {
-    const choiceSerial = $choice.data('serial');
+const _unsetGapFiller = function _unsetGapFiller(interaction, $gapFiller) {
+    const gapFillerSerial = $gapFiller.data('serial');
 
-    _choiceUsages[choiceSerial]--;
+    _gapFillerUsages[gapFillerSerial]--;
 
-    $choice.removeClass('disabled');
-    $choice.addClass('selectable');
-    $choice.find('img').removeAttr('draggable');
+    $gapFiller.removeClass('disabled');
+    $gapFiller.addClass('selectable');
+    $gapFiller.find('img').removeAttr('draggable');
+};
+
+/**
+ *
+ * @param {Object} interaction
+ * @param {JQuery} $fromGapFiller
+ * @param {Raphael.Element|JQuery} toElement
+ * @param {Function} endCallback
+ */
+const _animateMoveGapFiller = function _animateMoveGapFiller(interaction, $fromGapFiller, toElement, endCallback) {
+    const $container = containerHelper.get(interaction);
+    const $imageBox = $('.main-image-box', $container);
+    const boxOffset = $imageBox.offset();
+
+    const fromOffset = $fromGapFiller.offset();
+    const toOffset = toElement.paper ? $(toElement.node).offset() : toElement.offset();
+
+    const $img = $fromGapFiller.find('img');
+    const $clone = $img.clone();
+    $clone.css({
+        position: 'absolute',
+        display: 'block',
+        'z-index': 10000,
+        opacity: 0.8,
+        top: fromOffset.top - boxOffset.top,
+        left: fromOffset.left - boxOffset.left
+    });
+
+    $clone.appendTo($imageBox);
+    $clone.animate(
+        {
+            top: toOffset.top - boxOffset.top,
+            left: toOffset.left - boxOffset.left
+        },
+        200,
+        function animationEnd() {
+            $clone.remove();
+            endCallback();
```

Comment:
This line produces an error on unit01 when I click to remove a placed image.
<img width="519" height="338" alt="Screenshot 2025-12-29 at 11 44 34" src="https://github.com/user-attachments/assets/7a7f8023-ce00-491c-a61a-80b3198b3f1c" />